code/modules/mob/living/simple_animal/hostile/giant_spider.dm

```diff
@@ -200,56 +201,45 @@
 		for(var/mob/living/L in view(1,src))
 			if(L == src || L.anchored)
 				continue
+			if(istype(L, /mob/living/simple_animal/hostile/poison/giant_spider))
+				continue
 			if(Adjacent(L))
 				choices += L
 		for(var/obj/O in src.loc)
 			if(O.anchored)
 				continue
 			if(Adjacent(O))
 				choices += O
-		cocoon_target = input(src,"What do you wish to cocoon?") in null|choices
+		var/temp_input = input(src,"What do you wish to cocoon?") in null|choices
+		if(temp_input && !cocoon_target)
+			cocoon_target = temp_input
 
-	if(stat != DEAD && cocoon_target && busy != SPINNING_COCOON)
-		if(cocoon_target.anchored)
-			cocoon_target = null
-			return
+	if(stat != DEAD && cocoon_target && Adjacent(cocoon_target) && !cocoon_target.anchored)
+		if(busy == SPINNING_COCOON)
+			return //we're already doing this, don't cancel out or anything
 		busy = SPINNING_COCOON
-		src.visible_message("<span class='notice'>\the [src] begins to secrete a sticky substance around \the [cocoon_target].</span>")
+		visible_message("<span class='notice'>\the [src] begins to secrete a sticky substance around \the [cocoon_target].</span>")
 		stop_automated_movement = 1
 		walk(src,0)
-		if(do_after(src, 50, target = src))
+		if(do_after(src, 50, target = cocoon_target))
 			if(busy == SPINNING_COCOON)
-				if(cocoon_target && isturf(cocoon_target.loc) && get_dist(src,cocoon_target) <= 1)
-					var/obj/structure/spider/cocoon/C = new(cocoon_target.loc)
-					var/large_cocoon = 0
-					C.pixel_x = cocoon_target.pixel_x
-					C.pixel_y = cocoon_target.pixel_y
-					for(var/obj/item/I in C.loc)
-						I.forceMove(C)
-					for(var/obj/structure/S in C.loc)
-						if(!S.anchored)
-							S.forceMove(C)
-							large_cocoon = 1
-					for(var/obj/machinery/M in C.loc)
-						if(!M.anchored)
-							M.forceMove(C)
-							large_cocoon = 1
-					for(var/mob/living/L in C.loc)
-						if(istype(L, /mob/living/simple_animal/hostile/poison/giant_spider))
-							continue
-						large_cocoon = 1
-						L.forceMove(C)
-						C.pixel_x = L.pixel_x
-						C.pixel_y = L.pixel_y
+				var/obj/structure/spider/cocoon/C = new(cocoon_target.loc)
+				if(isliving(cocoon_target))
+					var/mob/living/L = cocoon_target
+					if(L.blood_volume && (L.stat != DEAD || !(L in consumed_mobs))) //if they're not dead, you can consume them anyway

```

Comment:
assoc list for ```consumed_mobs````, ```X in Y``` is unacceptably slow, especially when it's avoidable (like here)